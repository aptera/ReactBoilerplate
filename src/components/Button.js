import React, {Component} from 'react';

class Button extends Component {
    constructor(props) {
        super(props);
        this.state = {
            text: props.text
        }
        this.onHandleClick = props.click;
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        this.onHandleClick();
    }

    render() {
        return( 
            <button onClick={this.handleClick} type="button">{this.state.text}</button>
        );
    }
}

export default Button;