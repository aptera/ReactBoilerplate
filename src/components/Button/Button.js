import React, {Component} from 'react';
import './styles/button.css';

class Button extends Component {
    constructor(props) {
        super(props);
        this.state = {
            text: props.text
        };
        this.onHandleClick = props.click;
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        this.onHandleClick();
    }

    render() {
        return( 
            <button className="button" onClick={this.handleClick} type="button">{this.props.text}</button>
        );
    }
}

export default Button;