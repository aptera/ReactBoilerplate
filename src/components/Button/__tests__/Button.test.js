jest.unmock('../Button');
import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

describe('the avatar component', () => {
    describe('given a user name', () => {
        it('should display the users avatar', () => {

            let props = {
                text: 'My Test Button',
                click: () => null
            };

            const tree = renderer.create(
                <Button {...props} />
            ).toJSON();
            expect(tree).toMatchSnapshot();
            tree.props.onClick();
            expect(tree).toMatchSnapshot();
        });
    });
});