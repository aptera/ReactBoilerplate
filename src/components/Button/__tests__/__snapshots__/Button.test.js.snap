// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the avatar component given a user name should display the users avatar 1`] = `
<button
  className="button-styles"
  onClick={[Function]}
  type="button"
>
  My Test Button
</button>
`;

exports[`the avatar component given a user name should display the users avatar 2`] = `
<button
  className="button-styles"
  onClick={[Function]}
  type="button"
>
  My Test Button
</button>
`;
